dislib/data/classes.py

```diff
@@ -173,6 +186,13 @@ def _update_samples(self):
         for subset in self._subsets:
             self._samples = np.concatenate((self._samples, subset.samples))
 
+    def _update_samples_sparse(self):
+        self.collect()
+        self._samples = csr_matrix((0, self.n_features))
```

Comment:
I've noted that initializing an empty `_samples` causes the result to lose the information about the original `dtype`  (both in sparse and dense). I've changed it in my branch, so no need to do it here. Just letting you know

Reply:
Ok, good to know